plugins/woocommerce/src/Internal/Caches/EntityVersionsCache.php

```diff
@@ -0,0 +1,215 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Automattic\WooCommerce\Internal\Caches;
+
+use Automattic\WooCommerce\Proxies\LegacyProxy;
+
+/**
+ * Entity versions cache class.
+ *
+ * Provides a generic mechanism for caching versions of mutable entities.
+ */
+class EntityVersionsCache {
```

Comment:
Just out of curiosity, what do you think the benefits of doing this in a concrete class vs an abstraction where we create individual entity caches is?  E.g., `ProductVersionsCache extends EntityVersionsCache`?

The latter may give us a little bit more flexibility.

Reply:
[Your entity invalidation interface PR](https://github.com/woocommerce/woocommerce/pull/61613) could pair nicely with that inheritance scheme. Rather than having an invalidation interface/class, the `ProductVersionsCache` would register all of those hooks and bust the cache for products.